src/libraries/System.Net.Sockets/src/System/Net/Sockets/MultipleConnectAsync.cs

```diff
@@ -239,7 +239,10 @@ private void InternalConnectCallback(object? sender, SocketAsyncEventArgs args)
 
                 SocketAsyncEventArgs args = _internalArgs!;
                 args.RemoteEndPoint = new IPEndPoint(attemptAddress, _endPoint!.Port);
-                if (!attemptSocket.ConnectAsync(args))
+                bool pending = attemptSocket.ConnectAsync(args);
+                // Copy the socket address so we can use it in exception message.
+                _userArgs!._socketAddress = _internalArgs!._socketAddress;
```

Comment:
Is there reason why we don't use user supplied data? If we were connecting to name resolving to multiple addresses this would log just one, right?